catalog_plugin/api/catalog_api_client.py

```diff
@@ -66,7 +73,14 @@ def fetch_flexible_catalog(self):
             FlexibleCatalogModel or QuerySet.none(): The retrieved flexible catalog object or empty QuerySet if not found.
         """
         try:
-            if self.catalog_uuid:
+            if self.catalog_uuid_list:
+                catalogs = FlexibleCatalogModel.objects.filter(id__in=self.catalog_uuid_list)
+                if catalogs.exists():
```

Comment:
@Nekenhei To improve readability:

```suggestion
               if not catalogs.exists():
                  logger.warning(
                        f'No FlexibleCatalogModels found for any of the provided catalog_uuid_list: {self.catalog_uuid_list}.',
                    ) 
                return catalogs
```